### PR TITLE
Pass settings through levels

### PR DESCRIPTION
**Pull request by AntonioLangiu**
The idea of this PR is to pass the settings variable from the connection functions to the dns resolution function, so we are able to set options for the different levels with just one dictionary.

### CODE REVIEW

File: test/net/connect.cpp [hunk 1]
```diff
@@ -175,8 +175,7 @@ TEST_CASE("connect_first_of works with empty vector") {
                     REQUIRE(errors.size() == 0);
                     REQUIRE(bev == nullptr);
                     break_loop();
-                },
-                3.14);
+                });
```
Comment (bassosimone) [inline test/net/connect.cpp]: Here you remove timeout, right? If so, why?

**Comment by bassosimone** [comment]
Uhm, this is probably not ready for review yet. In any case, it looks a promising piece of work that simplifies matters and rationalizes the library. I like what I see.

File: test/net/connect.cpp (hunk 1 above)
Comment (AntonioLangiu) [inline test/net/connect.cpp]: No reason, sorry 👍  Good catch :)

**Comment by AntonioLangiu** [comment]
I force pushed on this branch to rebase it on top of refactor/settings_tags

**Pull request merged**